### PR TITLE
choose default deployment target for cluster if none is provided

### PR DESCRIPTION
**Pull request by d-g-town**

### CODE REVIEW

File: api/client/porter_app.go [hunk 1]
```diff
@@ -351,22 +351,33 @@ func (c *Client) DefaultDeploymentTarget(
...
+// CurrentAppRevisionInput is the input struct to CurrentAppRevision
+type CurrentAppRevisionInput struct {
+	ProjectID uint
+	ClusterID uint
+	AppName   string
+	// DeploymentTargetName is the name of the deployment target to get the current app revision for. One of this or DeploymentTargetID must be set.
+	DeploymentTargetName string
+	// DeploymentTargetID is the id of the deployment target to get the current app revision for. One of this or DeploymentTargetName must be set.
+	DeploymentTargetID string
+}
```
Comment (ianedwards) [inline api/client/porter_app.go:363]: is this unused?

File: cli/cmd/commands/app.go [hunk 2]
```diff
@@ -268,17 +268,23 @@ func appRollback(ctx context.Context, _ *types.GetAuthenticatedUserResponse, cli
 }
 
 func appRun(ctx context.Context, _ *types.GetAuthenticatedUserResponse, client api.Client, cliConfig config.CLIConfig, ff config.FeatureFlags, _ *cobra.Command, args []string) error {
+	deploymentTargetName, err := v2.ResolveDeploymentTargetNameFromFlag(ctx, client, cliConfig.Project, cliConfig.Cluster, deploymentTarget)
+	if err != nil {
+		return fmt.Errorf("could not resolve deployment target: %w", err)
+	}
```
Comment (ianedwards) [inline cli/cmd/commands/app.go:274]: high-level comment is why not just resolve this server side? specifically thinking about API users who might want to leave deployment target out safely

File: cli/cmd/v2/deployment_target.go [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import (
+	"context"
+	"fmt"
+
+	api "github.com/porter-dev/porter/api/client"
+)
+
+// ResolveDeploymentTargetNameFromFlag is a helper function for resolving the deployment target name from the flag, or defaulting to the default deployment target for the cluster
+func ResolveDeploymentTargetNameFromFlag(ctx context.Context, client api.Client, projectID uint, clusterID uint, flaggedDeploymentTargetName string) (string, error) {
+	targetID := flaggedDeploymentTargetName
```
Comment (ianedwards) [inline cli/cmd/v2/deployment_target.go:12]: `targetName`?

**Review by ianedwards — APPROVED** [review]

File: cli/cmd/v2/deployment_target.go (hunk 3 above)
Comment (ferozemohideen) [inline cli/cmd/v2/deployment_target.go:12]: targetName?

**Review by ferozemohideen — APPROVED** [review]

File: cli/cmd/commands/app.go (hunk 2 above)
Comment (d-g-town) [reply cli/cmd/commands/app.go:274]: duh, let me do that, thank you!

File: api/client/porter_app.go (hunk 1 above)
Comment (d-g-town) [reply api/client/porter_app.go:363]: is what unused?

**Pull request merged**